app/controllers/users_controller.rb

```diff
@@ -16,7 +15,25 @@ def create
     end
   end
 
-  def show
+  def edit
+    if current_admin? && User.find(params[:id]) == current_user
+      @user = current_user
+    else
+      render file: 'app/views/errors/not_found.html.erb'
+    end
+  end
+
+  def update
+    @user = current_user
+    # byebug
+    if @user.authenticate(params[:current_password])
```

Comment:
The user won't authenticate for some reason. I byebugged in and when I run this line it says false, but when I do @user.authenticate("password") it passes which is strange because params[:current_password], is literally "password". It should be the same check, but it fails for some reason.